CRM/Financeexportformats/Hook/BuildForm/FinancialBatchExport.php

```diff
@@ -19,10 +19,20 @@ public function run() {
   }
 
   /**
-   * Adds extra financial batch export formats
-   * to the exisitng core ones.
+   * Adds extra financial batch export formats to the exisitng core ones.
    */
   private function addExtraExportFormats() {
+
+    if (!empty($_GET['export_format'])
+    && in_array($_GET['export_format'], ['IIF', 'CSV', 'QuickbooksCSV'])
```

Comment:
As I mentioned in my other review, $_GET should be replaced with something more safe, even if just compare it with predefined array. 

also for better readability, this code can be extract into its own private method with a docblock that explains its purpose, , like this :


```
public function run() {
  $this->processQuickbooksFormatFromExportModalContext();
  $this->addExtraExportFormats();
}


  /**
   * CiviCRM versions < 5.39.1 has two ways to export batches, one of them
   * shows the export formats in a separate form, while the other shows them
   * in a Modal window, this adds Quickbooks format support to the Modal
   * window, and allow it to be processed from such context.
   */
private function processQuickbooksFormatFromExportModalContext() {
    $selectedExportFormat = $_GET['export_format'];
    if (empty($selectedExportFormat)
    || !in_array($selectedExportFormat, ['IIF', 'CSV', 'QuickbooksCSV'])
    ) {
        return;
     }

      $reflection = new \ReflectionProperty(get_class($this->form), '_exportFormat');
      $reflection->setAccessible(TRUE);
      $reflection->setValue($this->form, $_GET['export_format']);
      $this->form->postProcess();
}
```

Reply:
Hi @omarabuhussein,

Its thought lets keep this in extension code and add logic to check the CiviCRM version and put this hook calling logic behind the CiviCRM version check. 

that way we don't need to keep logic only for ASE.

Reply:
@kuldip-compuco 

Part of me wants to say I don't prefer that, because I don't want to give an indication that this extension supports CiviCRM versions < 5.39.1 , but I guess if we ended up adding it only for ASE, it might gets overwritten by someone else as part of extension upgrade or something, so yeah let's go with your suggestion.